airbyte-integrations/connectors/source-zendesk-talk/source_zendesk_talk/api.py

```diff
@@ -148,6 +148,49 @@ def read(self, getter: Callable) -> Iterator:
         yield from self._paginator(getter)
 
 
+class IncrementalStream(Stream, ABC):
+    """Stream that supports state and incremental read"""
+
+    state_pk = "timestamp"
+
+    @property
+    @abstractmethod
+    def updated_at_field(self):
+        """Name of the field associated with the state"""
+
+    @property
+    def state(self) -> Optional[Mapping[str, Any]]:
+        """Current state, if wasn't set return None"""
+        if self._state:
+            return {self.state_pk: str(self._state)}
+        return None
+
+    @state.setter
+    def state(self, value):
+        # Add 1 second so that when getting records are not duplicated with existing ones
+        self._state = pendulum.parse(value[self.state_pk]).add(seconds=1)
+        self._start_date = max(self._state, self._start_date)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._state = None
```

Comment:
just to confirm, this is being initialized with the latest state by BaseClient, right?

Reply:
yeah, of course